server/src/scanner/scanner.ts

```diff
@@ -177,8 +195,14 @@ export class Scanner {
           case ScanKind.Region:
             regions.push(result.result);
             break;
-          case ScanKind.Whitespace:
+          case ScanKind.Type:
+            // Were not able to reference earlier to lookback
+            // Gets first identifier on the line where type is defined
+            result.result.ref = tokens.filter(token => token.type === TokenType.identifier && token.start.line == result.result.start.line)[0];
```

Comment:
It seems as long as the array we are searching from (token array) does not grow too big, then it should not be a problem
https://github.com/dg92/Performance-Analysis-JS

On my biggest script I have token array of around 6000 elements.

Reply:
Maybe we just `find` instead of filter with index 0. I think its more clear that we're just looking for one element plus in the repo you linked to shows better performance so that seems like a win win.

Reply:
Yeap, changed it to find.